src/middleware/authentication.ts

```diff
@@ -0,0 +1,22 @@
+import { AuthenticationError } from "apollo-server";
+import { logger } from "../logger";
+import * as dotenv from "dotenv";
+
+dotenv.config();
+
+export const authenticationContextMiddleware = ({ req }: { req: any }) => {
+  const authHeaderUser: string = req.headers.authorization || "";
+  const expectedHeader: string = process.env.AUTH_TOKEN!;
+
+  if (authHeaderUser !== expectedHeader) {
+    logger.error(`Unauthorized access attempt: ${req.method} ${req.url}`);
```

Comment:
Usually logs cost money and we tend to avoid logging every request. Instead, we generate arbitrary ids which helps us trace requests down the line.

Reply:
I didn't understand what you mean exactly, can you give me some more explanations.